src/wallet3/keyring.hpp

```diff
@@ -109,19 +110,26 @@ namespace wallet
         PendingTransaction& ptx
     );
 
+    virtual std::vector<crypto::public_key> get_subaddress_spend_public_keys(uint32_t account, uint32_t begin, uint32_t end);
+
+    virtual void
+    expand_subaddresses(const cryptonote::subaddress_index& lookahead);
+
     virtual cryptonote::account_keys
     export_keys();
 
+    cryptonote::network_type nettype;
+
    private:
     crypto::secret_key spend_private_key;
     crypto::public_key spend_public_key;
 
     crypto::secret_key view_private_key;
     crypto::public_key view_public_key;
 
-    cryptonote::network_type nettype;
 
     hw::core::device_default key_device;
+    std::unordered_map<crypto::public_key, cryptonote::subaddress_index> subaddresses;
```

Comment:
thinking this should live in the database, but this is probably fine for now

Reply:
added a TODO